### PR TITLE
fix(frontend): fix analyzer results in trace DAG

### LINKED ISSUE #2638: Bug with analyzer error
Analyzer errors don't disappear when you click on next one

Image

- we don't need to have X icon on it at all, user click on error icon on span - see Analyzer errors, when he make another click somewhere, Analyzer errors must disappear 
- need to do something with graph alignment, when user click on error icon on span analyzer error not visible for 100%, look on the screen:

Image

### LINKED ISSUE #2639: Improve search input and graph positioning
Need improve search input and graph positioning (need to bring graphic to the back), see current example:

Image
Image
Image

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR includes multiples fixes/improvements for the Trace Analyzer results in the DAG:
- one single instance of the `Analyzer errors` component
- fix close behavior
- fix positioning in the DAG
- add connector line between `Analyzer errors` component and the `Span node`

## Changes

- one single instance of the `Analyzer errors` component
- fix close behavior
- fix positioning in the DAG
- add connector line between `Analyzer errors` component and the `Span node`

## Fixes

- fixes #2638 
- fixes #2639 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Loom

https://www.loom.com/share/40bc665d8ec3484b9b168d8706666150

### CODE REVIEW

**Comment by olha23** [comment]
Thank you @jorgeepc for the update, looks good

File: web/src/components/Visualization/components/DAG/SpanNode.tsx [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+import SpanService from 'services/Span.service';
+import {INodeDataSpan} from 'types/DAG.types';
+import AnalyzerResults from './AnalyzerResults';
+import * as SA from './AnalyzerResults.styled';
 import * as S from './SpanNode.styled';
 
 interface IProps extends NodeProps<INodeDataSpan> {}
 
 const SpanNode = ({data, id, selected}: IProps) => {
+  const dispatch = useAppDispatch();
```
Comment (xoscar) [inline web/src/components/Visualization/components/DAG/SpanNode.tsx:28]: We need to consider cleaning up nodes to not have to be connected to all of this state and such, maybe something to think latter on

**Review by xoscar — APPROVED** [review]
Hey @jorgeepc great job man!

File: web/src/components/Visualization/components/DAG/SpanNode.tsx (hunk 1 above)
Comment (jorgeepc) [reply web/src/components/Visualization/components/DAG/SpanNode.tsx:28]: Yes, agree with that. I have been thinking about that for a while. We need to parse everything (span related) and inject it in the data object.

**Pull request merged**